### PR TITLE
add fault/clues throughout exchange backup

### PR DESCRIPTION
**Pull request by ryanfkeepers**
## Does this PR need a docs update or release note?

- [x] :no_entry: No 

## Type of change

- [x] :broom: Tech Debt/Cleanup

## Issue(s)

* #1970

## Test Plan

- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: src/internal/connector/exchange/contact_folder_cache.go [hunk 1]
```diff
@@ -26,13 +26,13 @@ func (cfc *contactFolderCache) populateContactRoot(
 ) error {
 	f, err := cfc.getter.GetContainerByID(ctx, cfc.userID, directoryID)
 	if err != nil {
-		return support.ConnectorStackErrorTraceWrap(err, "fetching root folder")
+		return clues.Wrap(err, "fetching root folder")
```
Comment (ashmrtn) [inline src/internal/connector/exchange/contact_folder_cache.go:29]: I think this still needs `ConnectorStackErrorTrace` to get details
```suggestion
		return clues.Wrap(support.ConnectorStackErrorTrace(err), "fetching root folder")
```

File: src/internal/connector/exchange/service_iterators.go [hunk 2]
```diff
@@ -63,8 +64,8 @@ func filterContainersAndFillCollections(
 	}
 
 	for _, c := range resolver.Items() {
-		if ctrlOpts.FailFast && errs != nil {
-			return errs
+		if errs.Failed() {
+			return errs.Err()
```
Comment (ashmrtn) [inline src/internal/connector/exchange/service_iterators.go:68]: a bit concerned here too about only returning one of possibly several errors. Seems like either this function should no longer return an error and rely on the in/out param `errs` or should return all errors in `errs`

**Review by ashmrtn — APPROVED** [review]

File: src/internal/connector/exchange/service_iterators.go (hunk 2 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/service_iterators.go:68]: I feel like there's some misunderstanding of the fault package behavior we need to clear up.  We can't lose or hide errors, except by ignoring their population when we handle errors upstream.  What we're doing here isn't deciding _what_ errors to return.  It's only a consideration of _when_ to stop processing and return.

File: src/internal/connector/exchange/contact_folder_cache.go (hunk 1 above)
Comment (ryanfkeepers) [reply src/internal/connector/exchange/contact_folder_cache.go:29]: I haven't gotten around to updating the api package yet (being downstream, it required cascading changes to all upstream callers to implement, which is why this PR went first), but the expectation is that the API will pack that data into the error for this caller by use of [gaph.ErrData()](https://github.com/alcionai/corso/blob/main/src/internal/connector/graph/errors.go#L185).

**Pull request merged**